### ISSUE TITLE
[feature]: expose searchParams as an object on Location

### ISSUE THREAD

**Issue by bennypowers** [vaadin/vaadin-router#498]
How nice would it be to have `searchParams: Record<string, string|string[]>` on `RouterLocation`?

Not a hard implementation either:

```ts
const searchParams =
  Object.fromEntries(
    new URLSearchParams(location.search)
      .entries()
    );
```

**Comment by abdonrd** [comment]
Fixed by #582?

**Comment by haijian-vaadin** [comment]
Yes, we decided not to use `Record<string, string|string[]>`, but just to wrap it to a `URLSearchParams `, as implemented in #582.
